pkg/client/unit.go

```diff
@@ -146,11 +147,11 @@ func (u *Unit) Type() UnitType {
 	return u.unitType
 }
 
-// Expected returns the expected state and config for the unit.
-func (u *Unit) Expected() (UnitState, UnitLogLevel, *proto.UnitExpectedConfig) {
+// Expected returns the expected state, log leve, features and config for the unit.
+func (u *Unit) Expected() (UnitState, UnitLogLevel, *proto.Features, *proto.UnitExpectedConfig) {
```

Comment:
This is just the Go client, only changes in the .proto file affect endpoint. That said this feature will not fully work unless we align with endpoint, cloudbeat, and APM as well.

Reply:
I'm also not sure the features makes sense as a per unit return value here, since the feature flags are not actually per unit. 

A separate `Features()` channel seems like a better approach, that way users of this don't need to filter out and separately handle the feature changes from the unit changes.

Reply:
that's what we were talking with anderson in the mornign. providing a separate channel to watch inside checkin loop

Reply:
Yep, what is done now I think breaks the client API contract https://github.com/elastic/elastic-agent-client/pull/56#discussion_r1098692116

We need to change this before we can merge it.